### PR TITLE
Setup frontend testing 

### LINKED ISSUE #2049: Setup frontend testing
Phantomjs on capybara is very slow for testing javascript, and it's also not possible to test the details of the React Component.

To find a frontend testing framework which works with React. ( Currently I am looking into Jest and enzyme ).

### PR DESCRIPTION
**Pull request by allenwq**
Fix #2049 

- Setup `jest` and `enzyme` to test React code.
- Examples of integration test ( redux + react + react-router ) and unit test
- Setup code coverage

#### Component unit test
Use enzyme `shallow` renderer to render the component, and [snapshot](https://facebook.github.io/jest/docs/snapshot-testing.html) the component's state in different inputs.
Example: `client/app/bundles/course/survey/pages/SurveyIndex/__test__/index.test.js`

#### Integration test
This is the test that guarantees the code is actually working. All components including redux (actions, reducers, ...) and react-router will be involved in the test.
The general idea of frontend integration test is to stub the server side API, then we test whether the code works when backend gives the expected output. This will require us to write and document the server side API in detail so that it can be easily stubbed. 
Example: `client/app/bundles/course/survey/__test__/index.test.js`

### CODE REVIEW

File: client/app/bundles/course/survey/__test__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const store = storeCreator({ surveys: {} });
+
+    const indexPage = mount(
+      <ProviderWrapper store={store}>
+        <Router routes={routes} history={history} />
+      </ProviderWrapper>
+    );
+
+    // Wait for api call
+    await sleep(1);
```
Comment (weiqingtoh) [inline client/app/bundles/course/survey/__test__/index.test.js]: Just a random question, is there another way to wait for the API call, through a promise or something?

Comment (allenwq) [inline client/app/bundles/course/survey/__test__/index.test.js]: the other way I an think of is to somehow hook the API endpoint and implement a counter there,  then here will be a while loop until the counter is 0.

**Comment by allenwq** [comment]
😄 Coverage merged

**Comment by weiqingtoh** [comment]
LGTM - should we add some notes on `Contributing.MD`?

**Comment by allenwq** [comment]
@weiqingtoh Yep, i am going to update contributing guide in a sperate PR.

**Pull request merged**